src/client/l10n/useCurrentLocale.ts

```diff
@@ -0,0 +1,26 @@
+import { matchPath, useLocation } from 'react-router-dom'
+import { LocaleLabel, locales } from './locales'
+import { localePathPattern } from './localePathPattern'
+
+type LocaleUrlParam = {
+  locale: LocaleLabel
+}
+
+const FALLBACK_LOCALE: LocaleLabel = 'se-en'
+
+const getLocaleFromPath = (path: string): LocaleLabel => {
+  const localeMatch = matchPath<LocaleUrlParam>(path, {
+    path: localePathPattern + '/*',
+  })
+
+  const localeFromPath = localeMatch?.params?.locale
+
+  return localeFromPath ? localeFromPath : FALLBACK_LOCALE
+}
+
+export const useCurrentLocale = () => {
+  const location = useLocation()
+  const locale = getLocaleFromPath(location.pathname)
+  const currentLocale = locales[locale]
+  return currentLocale
+}
```

Comment:
Could we clarify what we get from this hook, maybe rename it? For me the concept of a locale is just a string, but we return an object with all settings for a specific locale. Something like `currentLocaleSettings` or `currentLocaleObject` :)

Reply:
I wanted it to be called the same as the hook we currently have since it made it a tiny bit easier to replace. And according to [Wikipedia](https://en.wikipedia.org/wiki/Locale_(computer_software)) at least a *locale* is not only a string representing language/region but all "special variant preferences":
> In computing, a locale is a set of parameters that defines the user's language, region and any special variant preferences that the user wants to see in their user interface.

Reply:
My perception is a bit off it seems 😄 So I think it's when we mix the naming like: `const locale = getLocaleFromPath(location.pathname)` since that is just a string and not an object. Just looking at the names you might expect `locale` and `currentLocale` to be of the same type. But I don't have a better for it so maybe we shouldn't overthink this 😄

Reply:
Ah, I see. Hm, maybe I should just rename this `locale` on line 23.

Reply:
Maybe like this?
![image](https://user-images.githubusercontent.com/42962286/134484429-41f7f60f-2ec3-47b8-846f-2758764d833f.png)

Reply:
That works IMO! 🙌